src/AEG/ProofNode.ts

```diff
@@ -0,0 +1,77 @@
+import {AEGTree} from "./AEGTree";
+
+/**
+ * The structure representing a single node of the proof
+ * @author Anusha Tiwari
+ */
+export class ProofNode {
+    /**
+     * The AEG Tree representing the current state of the proof
+     */
+    public tree: AEGTree;
+
+    /**
+     * The AEG Tree representing the previous state of the proof
+     */
+    public next: ProofNode | null = null;
+
+    /**
+     * The AEG Tree representing the next state of the proof
+     */
+    public previous: ProofNode | null = null;
+
```

Comment:
We also have to store what inference rule was used to construct the current graph, you could also compute this by just looking at the trees, but this would be a lot of work, much easier to just store it. This is important for when we want to show the inference rule history.

Reply:
I was wondering, should there also be a "rule" on when the user copies something from draw mode and pastes it into proof mode? Like can I just put "pasted subgraph" to signify the action